src/nu/validator/htmlparser/impl/Tokenizer.java

```diff
@@ -3359,6 +3347,37 @@ private void ensureBufferSpace(int inputLength) throws SAXException {
                          * I'm ∉ I tell you.
                          */
                     }
+                    // XXX reorder point
+                case AMBIGUOUS_AMPERSAND:
+                    /*
+                     * Unlike the definition is the spec, we don't consume the
+                     * next input character right away when entering this state;
+                     * that's because our current implementation differs from
+                     * the spec in that we've already consumed the relevant
+                     * character *before* entering this state.
+                     * Also, our implementation of this state has no looping.
+                     * So we never stay in this state; instead, we always
+                     * transition out from it back to returnState.
+                     */
+                    state = returnState;
```

Comment:
This assignment hides the transition from whatever is tracking the transitions. (In the Gecko case, the View Source highlighter.)

This line should be:
```
state = transition(state, returnState, reconsume, pos);
```